modules/apps/roles/roles-admin-web/src/main/java/com/liferay/roles/admin/web/internal/dao/search/SegmentsEntrySearchContainerFactory.java

```diff
@@ -0,0 +1,152 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.roles.admin.web.internal.dao.search;
+
+import com.liferay.portal.kernel.dao.search.EmptyOnClickRowChecker;
+import com.liferay.portal.kernel.dao.search.SearchContainer;
+import com.liferay.portal.kernel.portlet.PortletURLUtil;
+import com.liferay.portal.kernel.search.BaseModelSearchResult;
+import com.liferay.portal.kernel.search.Field;
+import com.liferay.portal.kernel.search.QueryConfig;
+import com.liferay.portal.kernel.search.SearchContext;
+import com.liferay.portal.kernel.search.Sort;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.segments.service.SegmentsEntryLocalServiceUtil;
+
+import java.io.Serializable;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.portlet.RenderRequest;
+import javax.portlet.RenderResponse;
+
+/**
+ * @author Pei-Jung Lan
+ */
+public class SegmentsEntrySearchContainerFactory {
+
+	public static SearchContainer create(
+			RenderRequest renderRequest, RenderResponse renderResponse)
+		throws Exception {
+
+		SearchContainer searchContainer = new SearchContainer(
+			renderRequest,
+			PortletURLUtil.getCurrent(renderRequest, renderResponse), null,
+			"no-segments-were-found");
+
+		searchContainer.setId("segmentsEntries");
+
+		String orderByCol = ParamUtil.getString(
+			renderRequest, "orderByCol", "name");
+
+		searchContainer.setOrderByCol(orderByCol);
+
+		String orderByType = ParamUtil.getString(
+			renderRequest, "orderByType", "asc");
+
+		searchContainer.setOrderByType(orderByType);
+
+		String tabs3 = ParamUtil.getString(renderRequest, "tabs3", "current");
+
+		searchContainer.setRowChecker(
+			new EmptyOnClickRowChecker(renderResponse));
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)renderRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		LinkedHashMap<String, Object> params = new LinkedHashMap<>();
+
+		if (tabs3.equals("current")) {
+			long roleId = ParamUtil.getLong(renderRequest, "roleId");
+
+			params.put("roleIds", new long[] {roleId});
+		}
+
+		BaseModelSearchResult baseModelSearchResult =
+			SegmentsEntryLocalServiceUtil.searchSegmentsEntries(
+				_buildSearchContext(
+					themeDisplay.getCompanyId(),
+					ParamUtil.getString(renderRequest, "keywords"), params,
+					searchContainer.getStart(), searchContainer.getEnd(),
+					_getSort(orderByCol, orderByType, themeDisplay)));
+
+		searchContainer.setResults(baseModelSearchResult.getBaseModels());
+		searchContainer.setTotal(baseModelSearchResult.getLength());
+
+		return searchContainer;
+	}
+
+	private static SearchContext _buildSearchContext(
```

Comment:
I had to build the search context manually because we want to get segment entries in any sites. Let me know if this is wrong.